loom/graphics/gfxGraphics.cpp

```diff
@@ -241,7 +244,55 @@ void Graphics::endFrame()
 
     if(pendingScreenshot[0] != 0)
     {
-        //bgfx::saveScreenShot(pendingScreenshot);
+		SDL_ClearError();
+		SDL_Window* SDLWindow = gSDLWindow;
+		SDL_Renderer* SDLRenderer = gSDLRenderer;
+
+		// Create a BMP image and save it to the requested file location
+		// Original algorithm by neilf (http://stackoverflow.com/a/20233470)
+		SDL_Surface* saveSurface = NULL;
+		SDL_Surface* infoSurface = NULL;
+		infoSurface = SDL_GetWindowSurface(SDLWindow);
+		if (infoSurface == NULL) {
+			lmLog(gGFXLogGroup, "Unable to create info surface from window for screenshot generation", SDL_GetError());
+			return;
+		}
+
+		unsigned char * pixels = new (std::nothrow) unsigned char[infoSurface->w * infoSurface->h * infoSurface->format->BytesPerPixel];
```

Comment:
This should use lmNew and friends. Also don't use array initializer; C++ likes to call constructors on each item in the array.